src/BindPattern.jl

```diff
@@ -0,0 +1,288 @@
+
+# Persistent data that we use across different patterns, to ensure the same computations
+# are always represented by the same synthetic variables.
+struct BinderState
+    # The module containing the pattern, in which types appearing in the
+    # pattern should be bound.
+    mod::Module
+
+    # The bindings to be used for each intermediate computations.  This maps from the
+    # computation producing the value to the symbol for the temp holding that value.
+    assignments::Dict{FetchPattern,Symbol}
+
+    # Errors found while binding.  Will be reported when executing the matching construct
+    errors::Vector{String}
+
+    # Assertions that should be executed at runtime before the matching code.
+    assertions::Vector{Any}
+
+    function BinderState(mod::Module)
+        new(
+            mod,
+            Dict{FetchPattern,Symbol}(),
+            Vector{String}(),
+            Vector{Any}()
+            )
+    end
+end
+
+const saved_prefix = "saved#"
+function get_temp(state::BinderState, p::FetchPattern)
+    get!(state.assignments, p) do; gensym(); end
+end
+get_temp(state::BinderState, p::FetchBindingBoundPattern) = get_temp(p.variable)
+get_temp(p::Symbol) = Symbol(saved_prefix, p)
+
+is_possible_type_name(t) = false
+is_possible_type_name(t::Symbol) = Base.isidentifier(t)
+function is_possible_type_name(t::Expr)
+    t.head == :. &&
+        is_possible_type_name(t.args[1]) &&
+        t.args[2] isa QuoteNode &&
+        is_possible_type_name(t.args[2].value)
+end
+
+function bind_pattern!(
+    location::LineNumberNode,
+    source::Any,
+    input::Symbol,
+    state::BinderState,
+    assigned::ImmutableDict{Symbol,Symbol})
+    @assert !occursin("/src/", string(location.file))
```

Comment:
This fails when I try to use `Rematch2` in another project.

Reply:
Fixed.